src/lifecycle/index.tsx

```diff
@@ -609,12 +610,19 @@ export class Updater {
   }
 
   async undelegate(
-    tracks: number[]
+    tracks: number[],
+    unlockTarget: string
   ): Promise<Result<SubmittableExtrinsic<'promise', SubmittableResult>>> {
     const state = this.#stateAccessor();
     const api = await this.getApi(state);
     if (api) {
-      const txs = tracks.map((track) => undelegate(api, track));
+      let txs = tracks.map((track) => undelegate(api, track));
+      if (unlockTarget) {
+        const unlockTxs = tracks.map((track) =>
+          unlock(api, track, unlockTarget)
```

Comment:
Shouldn't we only unlock when no more delegation is left? Someone might want to only unlock one track and keep the other ones.

Reply:
locks are per track.  We should unlock all tracks to free the balance. I don't see the reason we might want to have a complicated flow here. If the concern is the tx fee this won't add up that much fee.

Reply:
Hmm ok makes sense. Could you add some comment clarifying that? I fear I will forget otherwise.